docs/source/developer_docs.md

```diff
@@ -28,10 +28,10 @@ python3 -m venv venv
 source venv/bin/activate
 ```
 
-Then, it may be necessary to upgrade `pip`, `setuptools`, and `wheel`:
+Then, it may be necessary to upgrade `pip`:
 
 ```
-pip install -U pip setuptools wheel
+pip install -U pip
```

Comment:
In my experience, it's a good idea to run this when using `venv`. In particular, `wheel` is not installed by default, so installing other packages can involve compilation even when wheels are available. Any reason to skip this?

Reply:
> In particular, wheel is not installed by default, so installing other packages can involve compilation even when wheels are available

I don't think that this is true. What makes you believe that? At least in my venv it's not needed. The package descriptions also does not suggest anything like that.

Regarding setuptools: We don't use it anymore, and given the abundance of build backends and the fact that a suitable build backend should be specified in `pyproject.toml` for each modern package, I see no specific reason for recommending to install/upgrade it. Do you have a concrete argument, why this still would be needed?

Reply:
At least on my system, Python's `venv` doesn't install `wheel` (I ran the following with Python 3.10.11):

```bash
$ python -m venv test_venv
$ source test_venv/bin/activate
$ pip list
Package    Version
---------- -------
pip        23.0.1
setuptools 65.5.0

[notice] A new release of pip is available: 23.0.1 -> 23.1.2
[notice] To update, run: pip install --upgrade pip
```

Then when I run `pip install pymor[full]`, I get

```
  DEPRECATION: jupyter_contrib_core is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  DEPRECATION: jupyter-contrib-nbextensions is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
```

Reply:
Ah, I actually did not know about `pip list`. These packages do not appear in `pip freeze`. Anyway, what I am saying is that the `wheel` package should not be needed. It should be required for building wheels from source, so, e.g., `mpi4py` should pull it, not us.

Instead of `pip install pymor[full]`, you should look into

```
pip install -U pip
pip install 'pymor[full-compiled] @ git+https://github.com/pymor/pymor'
```

This works fine on my machine with Python 3.10.

Reply:
Ok, interesting. I don't really understand what happened, but it seems to work fine without installing `wheel`.